LLama/Control/IGenerationControl.cs

```diff
@@ -0,0 +1,31 @@
+﻿using LLama.Abstractions;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace LLama.Control
+{
+    /// <summary>
+    /// Control the text generation of LLama Executors.
+    /// </summary>
+    public interface IGenerationControl
+    {
+        /// <summary>
+        /// Use the last output text to determine if the generation should stop.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="inferenceParams"></param>
+        /// <param name="lastOutputText"></param>
+        /// <returns></returns>
+        bool ShouldStopGeneration(LLamaContext context, IInferenceParams inferenceParams, string lastOutputText);
+
+        /// <summary>
+        /// Use the last output ids to determine if the generation should stop.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="inferenceParams"></param>
+        /// <param name="lastOutputIds"></param>
+        /// <returns></returns>
+        bool ShouldStopGeneration(LLamaContext context, IInferenceParams inferenceParams, IEnumerable<int> lastOutputIds);
```

Comment:
Can this accept a single token instead? That gives it finer control (i.e. it can choose exactly _which_ token to stop at) and also is more efficient (at the moment most call sites will only have one token and will need to allocate a single item array for it).

Reply:
(Maybe also add an extension method helper with the current signature, which just loops over all tokens calling `ShouldStopGeneration` for each one).

Reply:
I used `IEnumerable<int>` because in some conditions one word is composed by two or more tokens. However after some investigation, I think it's not a common stage because when it happens, it means the model has few data of a certain language. Therefore the output of the model can't be good enough. The best choice for users is to use a model trained with their languages. I'll change it to use a single Id. :)